group1/comm/static/comm_tool.js

```diff
@@ -562,20 +562,31 @@ $(document).ready(function(){
   mobile_nav.sidebar();
 
   $('form#file_upload').submit(function(event){
-    $.ajax({
-      url: 'http://' + server_host + ':3000/upload',
-      type: 'POST',
-      data: new FormData( this ),
-      processData: false,
-      contentType: false,
-      success: function(file_path){
-        var download_url = 'http://' + server_host + ':' + server_port + '/' + file_path;
-        var display_name = $('input#filename').val();
-        $('input#text').val('<a href="' + download_url + '">' + display_name + '</a>' );
-        display();
-        $('#inputmodal').modal('hide');
-      }
-    });
+    var fileName = $('input#filename').val();
+    console.log(fileName);
+    if(testNameValidation(fileName)){
+      $.ajax({
+        url: 'http://' + server_host + ':3000/upload',
+        type: 'POST',
+        data: new FormData( this ),
```

Comment:
I know you didn't write this code, but there should be no spaces between the parentheses and "this".  Instead of "FormData( this )," it should be "FormData(this),"